src/GithubFilesBundle/Service/GithubImport.php

```diff
@@ -81,53 +81,46 @@ public function parseGithubDocument($lsDocKeys, $lsItemKeys, $fileContent){
-                }else{
-                    $lsDoc->setCreator($this->getValue($lsDocKeys['creator'], $row));
-                }
-
-                $lsDoc->setTitle($this->getValue($lsDocKeys['title'], $row));
-
-                $lsDoc->setOfficialUri($this->getValue($lsDocKeys['officialSourceURL'], $row));
-                $lsDoc->setPublisher($this->getValue($lsDocKeys['publisher'], $row));
-                $lsDoc->setDescription($this->getValue($lsDocKeys['description'], $row));
-                $lsDoc->setVersion($this->getValue($lsDocKeys['version'], $row));
-                $lsDoc->setSubject($this->getValue($lsDocKeys['subject'], $row));
-                $lsDoc->setLanguage($this->getValue($lsDocKeys['language'], $row));
-                $lsDoc->setNote($this->getValue($lsDocKeys['notes'], $row));
-
-                $em->persist($lsDoc);
-                $em->flush();
-                continue;
-            }
+        //create the new LsDoc
+        $lsDoc->setUri('local:'.$lsDoc->getIdentifier());
 
-            $lsItem = $this->parseGithubStandard($lsDoc, $lsItemKeys, $row);
-            //create a new association between the item and the doc
+        if(empty($this->getValue($lsDocKeys['creator'], $content[0]))){
+            $lsDoc->setCreator('Imported from GitHub');
+        }else{
+            $lsDoc->setCreator($this->getValue($lsDocKeys['creator'], $content[0]));
         }
+
+        $lsDoc->setTitle($this->getValue($lsDocKeys['title'], $content[0]));
+        $lsDoc->setOfficialUri($this->getValue($lsDocKeys['officialSourceURL'], $content[0]));
+        $lsDoc->setPublisher($this->getValue($lsDocKeys['publisher'], $content[0]));
+        $lsDoc->setDescription($this->getValue($lsDocKeys['description'], $content[0]));
+        $lsDoc->setVersion($this->getValue($lsDocKeys['version'], $content[0]));
+        $lsDoc->setSubject($this->getValue($lsDocKeys['subject'], $content[0]));
+        $lsDoc->setLanguage($this->getValue($lsDocKeys['language'], $content[0]));
+        $lsDoc->setNote($this->getValue($lsDocKeys['notes'], $content[0]));
+
+        $em->persist($lsDoc);
+
+        for($i=1; $i < sizeof($content); $i++){
+            $row = $content[$i];
+            $lsItem = $this->parseCSVGithubStandard($lsDoc, $lsItemKeys, $row);
+        }
+
+        $em->flush();
     }
 
-    public function parseGithubStandard(LsDoc $lsDoc, $lsItemKeys, $data){
+    public function parseCSVGithubStandard(LsDoc $lsDoc, $lsItemKeys, $data){
         $lsItem = new LsItem();
         $em = $this->getEntityManager();
 
         $lsItem->setLsDoc($lsDoc);
         $lsItem->setFullStatement($this->getValue($lsItemKeys['fullStatement'], $data));
 
-        $lsItemIdentifier = Uuid::uuid5(Uuid::NAMESPACE_URL, $this->getValue($lsItemKeys['fullStatement'], $data))->toString();
-        $lsItem->setIdentifier($lsItemIdentifier);
-        $lsItem->setUri('local:'.$lsItemIdentifier);
+        $lsItem->setUri('local:'.$lsItem->getIdentifier());
```

Comment:
Same here, the Uri does not need to be set since the identifier was not changed from what __construct() did.